sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/agents/models/RequestOptions.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// 
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package com.azure.search.documents.agents.models;
+
+import com.azure.core.annotation.Fluent;
+import java.util.UUID;
+
+/**
+ * Parameter group.
+ */
+@Fluent
+public final class RequestOptions {
```

Comment:
Should hide this from the public API as we do in the base package clients and indexes clients. Also don't expose it in the public APIs as we allow x-ms-client-request-id to be set through custom HttpPipelinePolicys.